src/lib/helpers/decorators.tsx

```diff
@@ -0,0 +1,53 @@
+import * as Analytics from "../../analytics/analytics"
+
+// This heavily uses ideas and approaches from @autobind
+// https://github.com/andreypopp/autobind-decorator
+export default function eventHandler(analyticsConstant: string) {
+  // TODO: disable tslint arrow function rule
+  return function bindMethod(target: any, propertyKey: string, descriptor: PropertyDescriptor) {
+    let originalMethod = descriptor.value
```

Comment:
@mennenia The original method is actually also in `target`, so maybe the way to set the return promise type expectation you need to do something like:

```ts
function bindMethod(target: () => Promise<any>, propertyKey: string, descriptor: PropertyDescriptor) {
  ...
}
```